src/Workspaces/Core/Portable/Utilities/TaskQueue.cs

```diff
@@ -0,0 +1,99 @@
+        public Task ScheduleTask(string taskName, Action operation, CancellationToken cancellationToken = default)
+            => EndOperation(BeginOperation(taskName), ScheduleTaskInProgress(operation, cancellationToken));
+
+        public Task<T> ScheduleTask<T>(string taskName, Func<T> operation, CancellationToken cancellationToken = default)
+            => EndOperation(BeginOperation(taskName), ScheduleTaskInProgress(operation, cancellationToken));
+
+        public Task ScheduleTask(string taskName, Func<Task> operation, CancellationToken cancellationToken = default)
+            => EndOperation(BeginOperation(taskName), ScheduleTaskInProgress(operation, cancellationToken));
+
+        public Task<T> ScheduleTask<T>(string taskName, Func<Task<T>> operation, CancellationToken cancellationToken = default)
+            => EndOperation(BeginOperation(taskName), ScheduleTaskInProgress(operation, cancellationToken));
+
+        [PerformanceSensitive("https://developercommunity.visualstudio.com/content/problem/854696/changing-target-framework-takes-10-minutes-with-10.html", AllowCaptures = false)]
+        public Task ScheduleTaskInProgress(Action operation, CancellationToken cancellationToken = default)
+        {
+            lock (_gate)
+            {
+                var task = _latestTask.SafeContinueWith(_ => operation(), cancellationToken, TaskContinuationOptions.None, _taskScheduler);
+                _latestTask = task;
+                return task;
+            }
+        }
+
+        [PerformanceSensitive("https://developercommunity.visualstudio.com/content/problem/854696/changing-target-framework-takes-10-minutes-with-10.html", AllowCaptures = false)]
+        public Task<T> ScheduleTaskInProgress<T>(Func<T> operation, CancellationToken cancellationToken = default)
+        {
+            lock (_gate)
+            {
+                var task = _latestTask.SafeContinueWith(_ => operation(), cancellationToken, TaskContinuationOptions.None, _taskScheduler);
+                _latestTask = task;
+                return task;
+            }
+        }
+
+        [PerformanceSensitive("https://developercommunity.visualstudio.com/content/problem/854696/changing-target-framework-takes-10-minutes-with-10.html", AllowCaptures = false)]
+        public Task ScheduleTaskInProgress(Func<Task> operation, CancellationToken cancellationToken = default)
+        {
+            lock (_gate)
+            {
+                var task = _latestTask.SafeContinueWithFromAsync(_ => operation(), cancellationToken, TaskContinuationOptions.None, _taskScheduler);
+                _latestTask = task;
+                return task;
+            }
+        }
+
+        [PerformanceSensitive("https://developercommunity.visualstudio.com/content/problem/854696/changing-target-framework-takes-10-minutes-with-10.html", AllowCaptures = false)]
+        public Task<T> ScheduleTaskInProgress<T>(Func<Task<T>> operation, CancellationToken cancellationToken = default)
```

Comment:
As an internal type, CancellationToken should not be optional.  It's bad when we forget to pass it along :)

Reply:
I kept it as it used to be, but I agree it should be explicit.